MetovaTestKit/CellTesting/MTKTableViewTestCell.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  MTKTableViewTestCell.swift
+//  MetovaTestKit
+//
+//  Created by Nick Griffith on 11/29/18.
+//  Copyright © 2018 Metova. All rights reserved.
+//
+//  MIT License
+//
+//  Permission is hereby granted, free of charge, to any person obtaining
+//  a copy of this software and associated documentation files (the
+//  "Software"), to deal in the Software without restriction, including
+//  without limitation the rights to use, copy, modify, merge, publish,
+//  distribute, sublicense, and/or sell copies of the Software, and to
+//  permit persons to whom the Software is furnished to do so, subject to
+//  the following conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+//  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+//  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+//  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import XCTest
+
+extension UITableView {
+    
+    /// This method attempts to produce a table view cell from the table view at the specified index path.  It then attempts to cast the cell to the specified type.  If either of these two fail, this function will produce a test failure and the test block will not be executed.  If both pass, the resulting cell, cast to the specified type, will be passed into the testBlock where the developer can run further assertions on the cell.
+    ///
+    /// - Parameters:
+    ///   - cellType: Type of cell to cast to before passing into test block.
+    ///   - indexPath: IndexPath to grab a cell from.
+    ///   - file: The file the test is called from.
+    ///   - line: The line the test is called from.
+    ///   - testBlock: A block of code containing tests to run.  This block is not guaranteed to execute.  If it does not execute, this method will fail the test.
+    /// - Throws: Rethrows any error thrown by the test block.  Does not throw any errors on its own.
+    func testCell<T: UITableViewCell>(
+        cellType: T.Type = T.self,
+        at indexPath: IndexPath,
+        file: StaticString = #file,
+        line: UInt = #line,
+        test testBlock: (T) throws -> Void
+    ) rethrows {
+        guard let dataSource = dataSource else {
+            XCTFail("\(self) does not have a dataSource", file: file, line: line)
+            return
+        }
+        
+        let tableSectionCount = dataSource.numberOfSections?(in: self) ?? 1
```

Comment:
I know you were previously attempting to use the tableView's internal `cellForRowAt` method to access the cell. Though it wasn't working. Had you tried using the tableView's `numberOfSections` method here rather than proxying through the delegate? Though the default behavior is to return `1` if the `dataSource` isn't implemented, do you feel it's necessary to re-implement that here?

Reply:
I never figured out exactly what was causing the original implementation to have issues.  I tried a few different things and can't remember all of them exactly, but this approach worked.  And on the specific point of going through the tableView's `numberOfSections` instead of the delegate, I actually considered it and intentionally wrote it this way for the sake of consistency throughout the entire method.

I'd greatly prefer to simply call `tableView.cellForRow(at: indexPath)`, but because that was running into whatever unknown issues, I decided to consistently go through the delegate.

Reply:
I wonder if `tableView.cellForRow(at: indexPath)` isn't returning values for cells that aren't actively in memory or visible on the table. 

Just looked it up actually, it seems that this is the case ([source](https://stackoverflow.com/a/32748935/4001148)). If a cell isn't visible, `tableView.cellForRow(at:)` is expected to return `nil`. In this particular case, it would make sense that could be the case.

I feel like proxying through the delegate is checking for configuration over outcome, though it indeed doesn't make too much sense to be testing Apple's `UITableView` implementation. I'll leave this one to your digression.